pkg/controller/logcollector/logcollector_controller.go

```diff
@@ -369,13 +373,40 @@ func (r *ReconcileLogCollector) Reconcile(request reconcile.Request) (reconcile.
 		pullSecrets,
 		installation,
 		r.clusterDomain,
+		render.OSTypeLinux,
 	)
 
 	if err := handler.CreateOrUpdate(ctx, component, r.status); err != nil {
 		r.status.SetDegraded("Error creating / updating resource", err.Error())
 		return reconcile.Result{}, err
 	}
 
+	// Render a fluentd component for Windows if the cluster has Windows nodes.
+	hasWindowsNodes, err := hasWindowsNodes(r.client)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if hasWindowsNodes {
```

Comment:
Any reason to not just always render a Windows daemonset? If there are no windows nodes it just won't run pods anyway, right?

I'm thinking it would be easier than this approach, because for this to work we need to monitor nodes to see if windows nodes are ever added to the cluster after initial provisioning.

Reply:
Node updates happen frequently, and so will cause unnecessary reconciliation in a lot of scenarios. Probably not an terrible issue, but certainly not ideal.

Reply:
Hmm... I was thinking it might be weird to have the daemonset even running on windows. That is a good point about the node update frequency that I didn't consider though.

Reply:
it wouldn't run on non-windows nodes though, right? We just slap a node selector on each so that the linux daemonset only runs on linux nodes, and the windows daemonset only runs on windows nodes (if any exist).

If the cluster has no windows nodes, then the windows daemonset pod won't run at all.

Reply:
I should have been more specific and said `it might be weird to have the daemonset even running at all`.
I did try this and I got a status degraded which makes sense:
```
  - lastTransitionTime: "2021-01-12T02:38:35Z"
    message: DaemonSet "tigera-fluentd/fluentd-node-windows" is not yet scheduled
      on any nodes
    reason: Not all pods are ready
```

I poked around but didn't see a good place to workaround. I looked at status manager but it only keeps track of namespaced names.

Reply:
Casey, we might merge this for now, just to help unblock some testing. I can address any follow-up/review you have afterwards.